sphinx4-core/src/main/java/edu/cmu/sphinx/api/FeatureExtractor.java

```diff
@@ -0,0 +1,146 @@
+package edu.cmu.sphinx.api;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+import edu.cmu.sphinx.frontend.Data;
+import edu.cmu.sphinx.frontend.DataEndSignal;
+import edu.cmu.sphinx.frontend.DataProcessingException;
+import edu.cmu.sphinx.frontend.DoubleData;
+import edu.cmu.sphinx.frontend.FloatData;
+import edu.cmu.sphinx.frontend.FrontEnd;
+import edu.cmu.sphinx.frontend.util.StreamDataSource;
+import edu.cmu.sphinx.util.props.ConfigurationManager;
+
+/**
+ * Extracts features from input stream
+ * 
+ * @author Vladisav Jelisavcic
+ * 
+ */
+class FeatureExtractor {
+    private FrontEnd frontEnd;
+    private StreamDataSource audioSource;
+    private int featureLength = -1;
+
+    /** The logger for this class */
+    private static final Logger logger = Logger
+            .getLogger("edu.cmu.sphinx.api.FeatureExtractor");
+
+    /**
+     * Constructs a FeatureExtractor. 
+     * 
+     * @param cm
+     *            the configuration manager
+     * @param frontEndName
+     *            the name for the frontEnd to be used
+     * @param inputStream
+     * 			  data stream 
+     *            
+     * @throws IOException if error occurred
+     */
+    public FeatureExtractor(ConfigurationManager cm, String frontEndName, InputStream inputStream)
```

Comment:
Hi,

I fixed the issues you mentioned, can you please review changes I made?

Thanks,
VJ

On Sat, Sep 12, 2015 at 11:08 AM, Nickolay V. Shmyrev <
notifications@github.com> wrote:

> In sphinx4-core/src/main/java/edu/cmu/sphinx/api/FeatureExtractor.java
> <https://github.com/cmusphinx/sphinx4/pull/37#discussion_r39335845>:
>
> > +    private static final Logger logger = Logger
> > +            .getLogger("edu.cmu.sphinx.api.FeatureExtractor");
> > +
> > +    /**
> > +     * Constructs a FeatureExtractor.
> > +     *
> > +     * @param cm
> > +     *            the configuration manager
> > +     * @param frontEndName
> > +     *            the name for the frontEnd to be used
> > +     * @param inputStream
> > +     * 			  data stream
> > +     *
> > +     * @throws IOException if error occurred
> > +     */
> > +    public FeatureExtractor(ConfigurationManager cm, String frontEndName, InputStream inputStream)
>
> Yes, configuration must be hidden. Our high-level API hides
> default.config.xml in the same way.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cmusphinx/sphinx4/pull/37/files#r39335845>.
>